kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java

```diff
@@ -143,7 +143,14 @@ public KieServerImpl() {
                                                                                         containerManager,
                                                                                         this), "KieServer-ControllerConnect");
             connectToControllerThread.start();
-
+            if (Boolean.parseBoolean(currentState.getConfiguration().getConfigItemValue(KieServerConstants.CFG_SYNC_DEPLOYMENT, "true"))) {
+                logger.info("Containers were requested to be deployed synchronously, holding application start...");
+                try {
+                    connectToControllerThread.join();
+                } catch (InterruptedException e1) {
+                    logger.debug("Interrupt exception when waiting for deployments");
```

Comment:
@yurloc I agree, though I am not 100% sure how to deal properly with this thread.join to be app server friendly and still secure the requirement for this feature... any ideas are more than welcome ... and PRs of course